runes: Use next id from runes table not `runes_uniqueid` from `vars`

ShahanaFarooqui opened the pull request:
Changelog-Fixed: runes: use runes table `id` instead `runes_uniqueid` from `vars` because it returns incorrect unique id if rune/s migrated from datastore.

Issue (#6696):
The issue occurs when user saves the rune (created by commando-rune) in the datastore with v23.02 and then upgrades the DB with v23.08. This upgrade calls migrate_datastore_commando_runes function to migrate runes from datastore to runes table. But it does not increment runes_uniqueid of vars table. So when user creates a new rune, CLN gets old next_unique_id and fails with UNIQUE_ID constraint error.
This issue effects Umbrel's users and users who saved their runes from older version in the datastore.

Solution 1:
- Add another migration function to recalculate `runes_uniqueid` and update vars table.
- Update `runes_uniqueid` in vars table in `migrate_datastore_commando_runes` for users who are yet to update.

Solution 2:
- Do not use `runes_uniqueid`, instead calculate the `next_unique_id` from `runes` table.
- `id` can be auto generated for runes insert but we still need to maintain it in runes because:
        - It is also required by `rune_derive_start`
        - counter starts from 1 (not 0) if not explicitly passed to the query.

Implemented solution 2 as it is clean and future proof.

rustyrussell commented:
According to the changelog lines and lack of tests, this doesn't change anything? :)

The original purpose of the var was that we can have missing runes. Perhaps that's a premature optimization, though. If so, you should delete the now-unused variable.

ShahanaFarooqui commented:
> According to the changelog lines and lack of tests, this doesn't change anything? :)

Updated changelog from `None` to `Fixed`. Also copied bug details from the original issue to describe the problem.

 
> The original purpose of the var was that we can have missing runes. Perhaps that's a premature optimization, though. If so, you should delete the now-unused variable.

Deleted unused variable.

rustyrussell reviewed (approved):
Ack 63fc3786b3cc7685397de36c12cb37b86c286841
